tests/test_errors.py

```diff
@@ -1333,3 +1334,44 @@ def test_parse_error_on_creation(create_func: Any, arg: Any) -> None:
         GrammarParseError, match=re.escape("no viable alternative at input '${b'")
     ):
         create_func(arg)
+
+
+def test_cycle_when_iterating_over_parents() -> None:
+    c = OmegaConf.create({"x": {}})
+    x_node = c._get_node("x")
+    assert isinstance(x_node, DictConfig)
+    c._set_parent(x_node)
+    with pytest.raises(
+        OmegaConfBaseException,
+        match=re.escape("Cycle when iterating over parents of key `x`"),
+    ):
+        c._get_full_key("x")
+
+
+def test_get_full_key_failure_in_format_and_raise() -> None:
+    c = OmegaConf.create({"x": {}})
+    x_node = c._get_node("x")
+    assert isinstance(x_node, DictConfig)
+    # We create a cycle in the parent relationship that will trigger a RecursionError
+    # when trying to access `c.x`. This test verifies that this RecursionError is properly
+    # raised even if another exception occurs in `format_and_raise()` when trying to
+    # obtain the full key.
+    c._set_parent(x_node)
+
+    # For some reason in Python 3.6 the error message may vary depending on how this
+    # test is run (`nox -s coverage-3.6` vs `nox -s omegaconf-3.6`). As a result, we
+    # do not check the message.
```

Comment:
I remember having some bad time trying to test recursion in previous versions. iirc in 2.7 Python the whole process was terminating sometimes without a proper error.
I am okay with limiting this testing to 3.7+ if it's stable there.

Reply:
Yes seems stable in 3.7+.
The test is also always raising a `RecursionError` in 3.6, but it doesn't always have the same error message... so I'm just skipping the test of the message for 3.6.

Reply:
As I said, In the past I saw that RecursionError was sometimes really crashing the Python process so I am a bit concerned about testing it at all.
If this is stable we can go with it though.

Reply:
> If this is stable we can go with it though.

It's been stable in my tests so far. If it turns out not to be, it's easy to disable this test until a replacement is found.